geode-core/build.gradle

```diff
@@ -163,93 +165,159 @@ dependencies {
+
+  //jaxb is used by cluster configuration
+  implementation('javax.xml.bind:jaxb-api')
+
+  //jaxb is used by cluster configuration
+  implementation('com.sun.xml.bind:jaxb-impl')
+
+  //istack appears to be used only by jaxb, not in our code. jaxb doesn't
+  //declare this as required dependency though. It's unclear if this is needed
+  //Runtime
+  runtimeOnly('com.sun.istack:istack-commons-runtime') {
     exclude group: '*'
   }
 
-  compile('org.apache.commons:commons-lang3')
-  compile('commons-modeler:commons-modeler') {
-    exclude module: 'commons-digester'
+  //Commons lang is used in many different places in core
+  implementation('org.apache.commons:commons-lang3')
+  
+  //Commons modeler is used by the (deprecated) admin API
+  implementation('commons-modeler:commons-modeler') {
     exclude module: 'commons-logging-api'
     exclude module: 'mx4j-jmx'
     exclude module: 'xml-apis'
     ext.optional = true
   }
-  compile('io.micrometer:micrometer-core')
     
-  compile('it.unimi.dsi:fastutil')
-  compile('javax.mail:javax.mail-api') {
+  //micrometer is used for micrometer based metrics from geode geode
+  api('io.micrometer:micrometer-core')
+
+
+  //FastUtil contains optimized collections that are used in multiple places in core
+  implementation('it.unimi.dsi:fastutil')
+
+  //Mail API is used by the deprecated admin API
+  implementation('javax.mail:javax.mail-api') {
     ext.optional = true
   }
-  compile('javax.resource:javax.resource-api')
-  compile('mx4j:mx4j') {
+
+  //The resource-API is used by the JCA support.
+  api('javax.resource:javax.resource-api')
+
+
+  //MX4J is used by the old admin API
+  implementation('mx4j:mx4j') {
     ext.optional = true
   }
-  compile('mx4j:mx4j-remote') {
+
+  //MX4J remote is used by both the old admin API 
+  implementation('mx4j:mx4j-remote') {
     ext.optional = true
   }
-  compile('mx4j:mx4j-tools') {
+
+  //MX4J tools is used by the old admin API
+  implementation('mx4j:mx4j-tools') {
     ext.optional = true
   }
-  compile('net.java.dev.jna:jna')
 
-  compile('net.sf.jopt-simple:jopt-simple')
+  //JNA is used for locking memory and preallocating disk files.
+  implementation('net.java.dev.jna:jna')
+
+  //JOptSimple is used by gfsh. A couple of usages have leaked into DiskStore
+  implementation('net.sf.jopt-simple:jopt-simple')
 
-  compile('org.apache.logging.log4j:log4j-api')
-  compile('org.apache.logging.log4j:log4j-core')
+  //Log4j is used everywhere
+  implementation('org.apache.logging.log4j:log4j-api')
+  implementation('org.apache.logging.log4j:log4j-core')
 
+  //Jansi is used by the CLI (maybe? it is a runtime dependency)
   runtimeOnly('org.fusesource.jansi:jansi') {
     ext.optional = true
   }
-  runtimeOnly('org.slf4j:slf4j-api')
 
+  //This routes slf4j logs to log4j. Shiro and micrometer use slf4j
   runtimeOnly('org.apache.logging.log4j:log4j-slf4j-impl') {
     exclude module: 'slf4j-api'
     ext.optional = true
```

Comment:
Not sure its optional, since we are using micometer and shiro...

Reply:
What these implementation jars do is control where the logging goes. In this case slf4j logging will get routed to log4j. 

But a consumer of geode might not want that, if they are using slf4j logging themselves.

If this jar is not present, it just means the logging from micrometer and shiro might not go to the log4j log.